src/live/customize.rs

```diff
@@ -169,6 +175,28 @@ impl LiveInitrd {
         Ok(())
     }
 
+    pub fn network_nmstate(&mut self, path: &str) -> Result<()> {
+        if !self.features.live_initrd_network {
+            bail!("This OS image does not support customizing network settings.");
+        }
+        let net_state_str = read_to_string(path)?;
+        match NetworkState::new_from_json(net_state_str.as_str()) {
+            Ok(net_state) => match net_state.gen_conf() {
```

Comment:
These two `match`es look like they could be collapsed to two `.context(...)?` instead to handle errors. That'll help unindent things.

Reply:
Tried that first but the NmstateError is not implemented the expected trait.

Reply:
Yup, it [looks like](https://github.com/nmstate/nmstate/blob/base/rust/src/lib/error.rs) `NmstateError` doesn't implement `Error`.  Ideally it would do so, but if necessary you can wrap the error in `anyhow!()` here.

Reply:
Going to try to make the right thing and put a PR at Nmstate so we have the proper NmstateError type there.

Reply:
Created PR at nmstate implementing the trait https://github.com/nmstate/nmstate/pull/1912